smibhid/lib/config/config_management.py

```diff
@@ -0,0 +1,54 @@
+from ulogging import uLogger
+import lib.config.config_template as config_template
+import config
+
+class ConfigManagement:
+    """
+    Ensure config file contains all items in config_template and that the values are of the correct type.
+    Apply defaults from config_template where values are missing in config.
+    """
+    def __init__(self) -> None:
+        self.log = uLogger("ConfigManagement")
+        self.error_count = 0
+    
+    def configure_error_handling(self) -> None:
+        """
+        Register errors with the error handler for the configuration management module.
+        """
+        from lib.error_handling import ErrorHandler
+        self.error_handler = ErrorHandler("ConfigManagement")
+        self.errors = {
+            "CFG": "Config errors found, check logs"
+        }
+
+        for error_key, error_message in self.errors.items():
+            self.error_handler.register_error(error_key, error_message)
+        
+        return
+        
+    def check_config(self) -> int:
+        self.log.info("Checking config file")
+        
+        for key in dir(config_template):
+            if not hasattr(config, key):
+                self.log.error(f"Missing config item in config.py: {key}, setting default value: {getattr(config_template, key)}")
```

Comment:
I dont think it should be an error for not specifying a config value that has a default.

If there is no default or no specified value, then an error is appropriate

Reply:
I had to make all the errors here be errors to satisfy the default config of 2 = error, in case the log config was missing. This way there will always be an error in the log and means I don't set a default log setting to high that might cause performance issues in production. Did a lot of back and forth on this. Also all items have a default, even if it is "" or None and some of those defaults are breaking defaults.

Reply:
ok fair enough.